File: trello-ladder/palindrome.rb
# Given a string, write a function that returns true if it is a palindrome, and false if it is not. (A palindrome is a word that reads the same both forward and backward.)

def palindrome?(string)
  inc = 0
  dec = string.length - 1
  while inc != dec
    if string[inc] != string[dec]
      return false
    end
  inc += 1
  dec -= 1
  end
  true
end

p palindrome?("racecar")
p palindrome?("baloney")

File: trello-ladder/basic-string/palindrome.js
// Given a string, write a function that returns true if it is a palindrome, and false if it is not. (A palindrome is a word that reads the same both forward and backward.)

// Input: “racecar”
// Output: true

// Input: “baloney”
// Output: false

function isPalindrome(string) {
  var reverseIndex = string.length - 1;
  for (let index = 0; index < string.length; index++) {
    if (string[index] !== string[reverseIndex]) {
      return false;
    }
    reverseIndex--;
  }
  return true;
}

console.log(isPalindrome("racecar"));
console.log(isPalindrome("baloney"));
